provisioning/service/src/Authentication/ServiceConnectionString.cs

```diff
@@ -1,156 +1,235 @@
+        public static ServiceConnectionString Create(string serviceConnectionString)
         {
-            if (parser == null)
+            if (string.IsNullOrWhiteSpace(serviceConnectionString))
             {
-                throw new ArgumentNullException(nameof(parser));
+                throw new ArgumentNullException(nameof(serviceConnectionString));
             }
 
-            HostName = parser.HostName;
-            SharedAccessKeyName = parser.SharedAccessKeyName;
-            SharedAccessKey = parser.SharedAccessKey;
-            SharedAccessSignature = parser.SharedAccessSignature;
-            ServiceName = parser.ServiceName;
-            HttpsEndpoint = new UriBuilder("https", parser.HostName).Uri;
-        }
+            var ServiceConnectionStringParser = new ServiceConnectionString();        
+            ServiceConnectionStringParser.Parse(serviceConnectionString);
 
-        /// <summary>
-        /// The Provisioning Service Name
-        /// </summary>
-        public string ServiceName
-        {
-            get;
-            private set;
+            return ServiceConnectionStringParser;
         }
 
         /// <summary>
         /// The Provisioning Service Client Hostname
         /// </summary>
         public string HostName
         {
-            get;
-            private set;
+            get { return _hostName; }
+            set { SetHostName(value); }
         }
 
         /// <summary>
-        /// The Provisioning Service Client Https Endpoint
+        /// The Provisioning Service Access Policy Name
         /// </summary>
-        public Uri HttpsEndpoint
-        {
-            get;
-            private set;
-        }
+        public string SharedAccessKeyName { get; internal set; }
 
         /// <summary>
-        /// The Provisioning Service Audience
+        /// The Provisioning Service Shared Access Key for the specified
+        /// access policy
         /// </summary>
-        public string Audience
-        {
-            get { return HostName; }
-        }
+        public string SharedAccessKey { get; internal set; }
 
         /// <summary>
-        /// The Provisioning Service Access Policy Name
+        /// The Provisioning Service Shared Access Signature
         /// </summary>
-        public string SharedAccessKeyName
-        {
-            get;
-            private set;
-        }
+        public string SharedAccessSignature { get; internal set; }
 
         /// <summary>
-        /// The Provisioning Service Shared Access Key for the specified
-        /// access policy
+        /// The Provisioning Service Name
         /// </summary>
-        public string SharedAccessKey
+        public string ServiceName
         {
-            get;
-            private set;
+            get { return _serviceName; }
         }
 
         /// <summary>
-        /// The Provisioning Service Shared Access Signature
+        /// Returns the Provisioning Service Connection string
```

Comment:
Please correct capitalization. Also, end all sentences in "." (here and everywhere).
In general, for documentation please follow the MSDN guidelines or examples from the .NET Framework. The correct structure for properties is "Gets or sets ...".